### PR TITLE
tweaks to library panel

### PR DESCRIPTION
**Pull request by placegraphichere**
- create and rename is full width of panel
- input for rename/create color stays consistent color
- cancel/create/rename buttons have hover and style to match spectrum
spec
- confirm dialog tweaked to match spectrum spec: padding, font-size,
primary/secondary buttons and hover

added asset rollover color for text, changed order of icons

- Removed redundant strings, background covers 100% height

- focus state for editing input

align libraries bar with spec and Ps icon on the right

issue #2431 library renaming interaction conflicts

- adding input to library selected state
- double click still edits, but icons truncate the asset name
appropriately in the space
- added selected state for library input title

- fixed "leave library" string
- removed duplicate variable in assetSection.jsx code

### CODE REVIEW

**Comment by iwehrman** [comment]
This looks weird to me:
image

**Comment by placegraphichere** [comment]
adding review only, I'm gonna do a few small things to make layers and this more consistent with select/focus inputs

**Comment by placegraphichere** [comment]
@iwehrman what looks weird? the shadow on the back? If so, I agree

**Comment by iwehrman** [comment]
(Sorry, I pasted the wrong picture originally. Yes, I was referring to the shadow transparency.)

**Comment by iwehrman** [comment]
Also, it seems weird that we list the asset title twice. Department of redundancy department.

**Comment by iwehrman** [comment]
I realize the wording mirrors standard PS, but I still don't like it. Strawman:

Title: Delete Library (or Asset). [Note that that has a fixed length, which is good for titles.]
Body: Are you sure you want to delete Library (or Asset) "\<name\>"?

**Comment by davealonzo** [comment]
I agree with the redundancy of the title and body text. Could we consider changing the title to "Delete Library" and "Delete Library Asset", and just keep the <name> in the body?

**Comment by davealonzo** [comment]
Correction... "<name>" in the body

**Comment by davealonzo** [comment]
what Ian said ;-)

**Comment by placegraphichere** [comment]
screen shot 2015-09-14 at 1 07 00 pm
screen shot 2015-09-14 at 1 06 52 pm

**Comment by placegraphichere** [comment]
Ready for review now @shaoshing

File: src/nls/root/strings.json [hunk 1]
```diff
@@ -474,13 +474,13 @@
     },
     "LIBRARIES": {
         "CREATE_LIBRARY": "Create New Library",
-        "RENAME_LIBRARY": "Rename \"%s\"",
-        "DELETE_LIBRARY": "Delete \"%s\"",
+        "RENAME_LIBRARY": "Rename Library",
+        "DELETE_LIBRARY": "Delete Library",
         "DELETE_LIBRARY_CONFIRM": "Are you sure you want to delete \"%s\"?",
         "DELETE_SHARED_LIBRARY_CONFIRM": "Your collaborators will no longer have access to this library if you delete it. Are you sure you want to delete \"%s\"?",
         "LEAVE_LIBRARY": "Leave \"%s\"",
```
Comment (shaoshing) [inline src/nls/root/strings.json]: Found one more title  "Leave Library" 

screen shot 2015-09-14 at 11 30 04 am

**Comment by shaoshing** [comment]
The asset buttons seems not align with asset titles

screen shot 2015-09-14 at 11 24 24 am

File: src/js/jsx/sections/libraries/LibraryList.jsx [hunk 2]
```diff
@@ -153,12 +153,12 @@ define(function (require, exports, module) {
 
                 options = options.concat([
                     {
-                        title: strings.LIBRARIES.RENAME_LIBRARY.replace("%s", selectedLibrary.name),
```
Comment (shaoshing) [inline src/js/jsx/sections/libraries/LibraryList.jsx]: I love the new title, but we have the localized titles that still have the string tags. Maybe keep these replacement code until we confirm the translations have updated? @iwehrman 

https://github.com/adobe-photoshop/spaces-design/blob/master/src/nls/fr/strings.json#L472

**Comment by shaoshing** [comment]
@placegraphichere Found two nits, and a question about localized titles. Back to you.

**Comment by placegraphichere** [comment]
I added in issue #2431 with changes to the selected state for a library asset. This changes some behavior of selection so that it always shows the asset title. The title will then truncate with an ellipsis when the asset is selected to show the three icons for the layer now on the RIGHT side of the panel. 

This revised design has been shown to @iwehrman and @davealonzo for review before this PR. Here's the new interaction:

libraries-select-hover-rename

Review only at this point, back to @shaoshing on my question - review of AssetSection.jsx render() function changes

File: src/js/jsx/sections/libraries/assets/AssetSection.jsx [hunk 3]
```diff
@@ -187,10 +198,14 @@ define(function (require, exports, module) {
...
                             onClick={ui.openURL.bind(null, elementLink)} />
+                         <SplitButtonItem
+                            title={strings.TOOLTIPS.LIBRARY_DELETE}
+                            iconId="delete"
+                            onClick={this._handleDelete} />
                     </SplitButtonList>
                 );
             } else {
-                var subTitle = this.props.subTitle && (<div className="libraries__asset__subtitle">
+                subTitle = this.props.subTitle && (<div className="libraries__asset__subtitle">
```
Comment (shaoshing) [inline src/js/jsx/sections/libraries/assets/AssetSection.jsx]: This seems to be the same compared with line 170, we can safely remove these lines.

**Comment by shaoshing** [comment]
@placegraphichere nice work on the asset buttons tweak. there are two more nits. one is duplicate code, and another is the out dated "Leave %s" string.

**Comment by placegraphichere** [comment]
Back to you @shaoshing, thanks for being patient

**Comment by shaoshing** [comment]
Thank you @placegraphichere .

**Pull request merged**